### PR TITLE
Apply updates from core

### PR DESCRIPTION
**Pull request by joon9823**
- rename slinky module to connect
- add order param to MsgRegisterAccount
- update tally result from initia.gov.v1.Proposal
- add missing query/msgs

## Summary by CodeRabbit

- **New Features**
	- Introduced `MsgUpdateIbcPermAdmin` and `MsgUpdatePermissionedRelayers` for managing IBC permissions.
	- Added `MsgUpsertMarkets` class for handling market updates.

- **Enhancements**
	- Updated API endpoints across various modules to reflect new versioning.
	- Improved handling of message types with updated import paths and identifiers.

- **Bug Fixes**
	- Fixed inconsistencies in message type handling across multiple classes.

- **Chores**
	- Updated package dependencies and versioning in `package.json` (version 0.2.17).
	- Refactored import paths to align with new protocol definitions.

### CODE REVIEW

**Comment by joon9823** [comment]
all of the coderabbit comments don't seem valid or appropriate..

File: src/core/gov/TallyResult.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+  public static fromAmino(data: TallyResult.Amino): TallyResult {
+    const {
+      tally_height,
+      total_staking_power,
+      total_vesting_power,
+      v1_tally_result,
+    } = data
+    return new TallyResult(
+      Number.parseInt(tally_height),
```
[the remark this answers is not in the document]
Comment (Vritra4) [reply src/core/gov/TallyResult.ts:34]: not an issue

File: src/core/Msg.ts [hunk 2]
```diff
@@ -515,12 +515,10 @@ export namespace Msg {
...
-      case 'perm/MsgSetPermissionedRelayers':
-        return MsgSetPermissionedRelayers.fromAmino(data)
-      case 'perm/MsgHaltChannel':
-        return MsgHaltChannel.fromAmino(data)
-      case 'perm/MsgResumeChannel':
-        return MsgResumeChannel.fromAmino(data)
+      case 'ibc-perm/MsgUpdateAdmin':
+        return MsgUpdateIbcPermAdmin.fromAmino(data)
+      case 'ibc-perm/MsgUpdatePermissionedRelayers':
+        return MsgUpdatePermissionedRelayers.fromAmino(data)
```
[the remark this answers is not in the document]
Comment (joon9823) [reply src/core/Msg.ts:521]: this is the valid identifier

File: src/core/Msg.ts [hunk 3]
```diff
@@ -633,9 +633,9 @@ export namespace Msg {
         return MsgUpdateOphostParams.fromAmino(data)
 
       // oracle
-      case 'slinky/x/oracle/MsgAddCurrencyPairs':
+      case 'connect/x/oracle/MsgAddCurrencyPairs':
         return MsgAddCurrencyPairs.fromAmino(data)
-      case 'slinky/x/oracle/MsgSetCurrencyPairs':
+      case 'connect/x/oracle/MsgSetCurrencyPairs':
```
[the remark this answers is not in the document]
Comment (joon9823) [reply src/core/Msg.ts]: fixed

File: src/core/gov/TallyResult.ts [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+      tally_height: tally_height.toString(),
+      total_staking_power,
+      total_vesting_power,
+      v1_tally_result,
+    }
+  }
+
+  public static fromProto(data: TallyResult.Proto): TallyResult {
+    return new TallyResult(
+      data.tallyHeight.toNumber(),
```
[the remark this answers is not in the document]
Comment (joon9823) [reply src/core/gov/TallyResult.ts:88]: we are using number type for convenience so we want to stick to it

File: src/core/marketmap/msgs/MsgUpsertMarkets.ts [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    const {
+      value: { authority, markets },
+    } = data
+    return new MsgUpsertMarkets(authority, markets.map(Market.fromAmino))
+  }
+
+  public toAmino(): MsgUpsertMarkets.Amino {
+    const { authority, markets } = this
+    return {
+      type: 'connect/x/marketmap/MsgUpsertMarkets',
```
[the remark this answers is not in the document]
Comment (joon9823) [reply src/core/marketmap/msgs/MsgUpsertMarkets.ts:33]: this is from the original protobuf definition so this is correct

File: src/core/marketmap/msgs/MsgUpdateMarkets.ts [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import { JSONSerializable } from '../../../util/json'
 import { AccAddress } from '../../bech32'
 import { Any } from '@initia/initia.proto/google/protobuf/any'
-import { MsgUpdateMarkets as MsgUpdateMarkets_pb } from '@initia/initia.proto/slinky/marketmap/v1/tx'
+import { MsgUpdateMarkets as MsgUpdateMarkets_pb } from '@initia/initia.proto/connect/marketmap/v2/tx'
```
[the remark this answers is not in the document]
Comment (joon9823) [reply src/core/marketmap/msgs/MsgUpdateMarkets.ts:4]: verified that this is correct

File: src/core/marketmap/msgs/MsgUpsertMarkets.ts [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+  public static fromData(data: MsgUpsertMarkets.Data): MsgUpsertMarkets {
+    const { authority, markets } = data
+    return new MsgUpsertMarkets(authority, markets.map(Market.fromData))
+  }
+
+  public toData(): MsgUpsertMarkets.Data {
+    const { authority, markets } = this
+    return {
+      '@type': '/connect.marketmap.v2.MsgUpsertMarkets',
```
[the remark this answers is not in the document]
Comment (joon9823) [reply src/core/marketmap/msgs/MsgUpsertMarkets.ts:49]: verified that this is correct

File: src/core/marketmap/msgs/MsgUpsertMarkets.ts [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+  ) {
+    super()
+  }
+
+  public static fromAmino(data: MsgUpsertMarkets.Amino): MsgUpsertMarkets {
+    const {
+      value: { authority, markets },
+    } = data
+    return new MsgUpsertMarkets(authority, markets.map(Market.fromAmino))
+  }
```
[the remark this answers is not in the document]
Comment (joon9823) [reply src/core/marketmap/msgs/MsgUpsertMarkets.ts:28]: doesn't seem essential

File: src/core/marketmap/msgs/MsgUpsertMarkets.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (joon9823) [reply src/core/marketmap/msgs/MsgUpsertMarkets.ts:33]: this is from original protobuf definition so gotta stick to it

File: src/core/ophost/msgs/MsgUpdateChallenger.ts [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+    data: MsgUpdateChallenger.Amino
+  ): MsgUpdateChallenger {
+    const {
+      value: { authority, bridge_id, challenger },
+    } = data
+
+    return new MsgUpdateChallenger(
+      authority,
+      Number.parseInt(bridge_id),
+      challenger
```
[the remark this answers is not in the document]
Comment (joon9823) [reply src/core/ophost/msgs/MsgUpdateChallenger.ts:35]: doesn't seem essential

File: src/core/ophost/msgs/MsgUpdateChallenger.ts [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+  /**
+   * @param authority the address that controls the module
+   * @param bridge_id
+   * @param challenger
+   */
+  constructor(
+    public authority: AccAddress,
+    public bridge_id: number,
+    public challenger: AccAddress
+  ) {
```
[the remark this answers is not in the document]
Comment (joon9823) [reply src/core/ophost/msgs/MsgUpdateChallenger.ts:21]: doesn't seem essential

File: src/core/ibc/applications/perm/msgs/MsgUpdateIbcPermAdmin.ts [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+    const { authority, channel_id, port_id, admin } = this
+    return {
+      type: 'ibc-perm/MsgUpdateAdmin',
+      value: {
+        authority,
+        channel_id,
+        port_id,
+        admin,
+      },
+    }
```
[the remark this answers is not in the document]
Comment (joon9823) [reply src/core/ibc/applications/perm/msgs/MsgUpdateIbcPermAdmin.ts:45]: this is intended difference

**Review by Vritra4 — APPROVED** [review]
LGTM

**Pull request merged**